skyportal/handlers/api/catalog_services.py

```diff
@@ -0,0 +1,223 @@
+import base64
+import functools
+import glob
+import numpy as np
+import pandas as pd
+from sqlalchemy.orm import sessionmaker, scoped_session
+import sqlalchemy as sa
+import swifttools.ukssdc.query as uq
+import tempfile
+from tornado.ioloop import IOLoop
+
+from baselayer.app.access import auth_or_token
+from baselayer.app.env import load_env
+from baselayer.log import make_log
+
+from .source import post_source
+from .photometry import add_external_photometry
+from ..base import BaseHandler
+from ...models import (
+    Comment,
+    DBSession,
+    Group,
+    Obj,
+    Telescope,
+    User,
+)
+
+_, cfg = load_env()
+
+
+log = make_log('api/catalogs')
+
+Session = scoped_session(sessionmaker(bind=DBSession.session_factory.kw["bind"]))
+
+
+class SwiftLSXPSQueryHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: |
+            get Swift LSXPS objects and post them as sources
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  telescope_name:
+                    required: false
+                    type: integer
+                    description: |
+                      Name of telescope to assign this catalog too
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+
+        telescope_name = data.get('telescope_name', 'Swift')
```

Comment:
In what case would the telescope be something other than Swift?

Reply:
This is in case someone used a different name.

Reply:
If a user specifies a different name would it still fetch the same sources from that catalog? Would they just be posted under a different name?

Reply:
Yes exactly. But that instrument will also need the swiftxrt filter, so could not choose randomly.

Reply:
I don't get it. The catalog that is queried here is strictly sources from Swift.

Reply:
Yes. But what if they didnt name it Swift? :)

Reply:
I seriously don't understand when that could happen. Wouldn't it be easier to just assume this handler is for Swift only?

Reply:
Well they could still do Neil Gehrels Swift Observatory and such. But I can remove it if that makes sense to you.

Reply:
You mean if a skyportal admin runs their own database and calls the swift telescope something other than `Swift` you would not find that telescope on the DB?

Reply:
yes.

Reply:
OK now I understand. I think you can leave it in. Maybe document that in the docstring (something like "use the same name as the nickname given to the Swift telescope").